### PR TITLE
Fix support for doctrine/mongodb-odm 2.x

### LINKED ISSUE #152: EmbedOne: How to configureFormFields ?
When I use `->add('user.name')` I have an error: 
`No mapping found for field 'user__name' in class '..\Document\Account'.`

But the mapping is in the embed class, called in my case as EmbedUser ...

``` php
class Account
{
    /**
     * @MongoDB\Id
     */
    protected $id;

    /**
     * @MongoDB\EmbedOne(targetDocument="EmbedUser")
     */
    protected $user;
[...]
```

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

Using `ext-mongodb` with the version 2.x of `doctrine/mongodb-odm`, the `\MongoRegex` and `\MongoId` classes don't exist.

The option `simple` [was deprecated](https://www.doctrine-project.org/projects/doctrine-mongodb-odm/en/1.3/reference/annotations-reference.html#referenceone) and doesn't exist in 2.x of `doctrine/mongodb-odm`.

Edit: 

I changed from `.$id` to `._id`, because it didn't work to me, if I had understood it correctly, [`$id` is used when there is a reference](https://docs.mongodb.com/manual/reference/database-references/) and [for usual ids and embedded `_id`](https://docs.mongodb.com/manual/reference/method/db.collection.find/);

I am targeting this branch, because these changes respect BC.

Closes #152

## Changelog

```markdown
### Fixed
- Fixed support for doctrine/mongodb-odm 2.x
```

### CODE REVIEW

File: src/Filter/StringFilter.php [hunk 1]
```diff
@@ -72,4 +73,18 @@ public function getRenderSettings()
...
     }
+
+    /**
+     * NEXT_MAJOR: Use only Regex when dropping support for doctrine/mongodb-odm 1.x.
+     *
+     * @return Regex|\MongoRegex
+     */
+    private function getRegexExpression(string $pattern)
+    {
+        if (class_exists(Regex::class)) {
```
Comment (core23) [inline src/Filter/StringFilter.php]: Can you provide a test for this @franmomu ?

File: tests/Filter/ModelFilterTest.php [hunk 2]
```diff
@@ -205,12 +207,55 @@ public function testAssociationWithValidParentAssociationMappings()
...
 
         $filter->apply($builder, ['type' => EqualType::TYPE_IS_EQUAL, 'value' => new DocumentStub()]);
 
         $this->assertTrue($filter->isActive());
     }
+
+    /**
+     * @dataProvider getMappings
+     */
+    public function testDifferentIdentifiersBasedOnMapping(string $storeAs, string $fieldIdentifier)
```
Comment (phansys) [inline tests/Filter/ModelFilterTest.php]: Please, add the missing return declarations.

File: src/Filter/ModelFilter.php [hunk 3]
```diff
@@ -15,6 +15,9 @@
 
 use Doctrine\Bundle\MongoDBBundle\Form\Type\DocumentType;
 use Doctrine\Common\Collections\Collection;
+use Doctrine\ODM\MongoDB\Mapping\ClassMetadata;
+use MongoDB\BSON\ObjectId;
```
Comment (OskarStark) [inline src/Filter/ModelFilter.php]: Looks unused in this scope, you are using \ObjectId

Comment (franmomu) [reply src/Filter/ModelFilter.php]: I used \ObjectId in PHPDoc, but it was a mistake, this is the class: https://www.php.net/manual/en/class.mongodb-bson-objectid.php

Comment (OskarStark) [reply src/Filter/ModelFilter.php]: Could you the fix the PHPDoc please, and also check the other files? Thanks

Comment (franmomu) [reply src/Filter/ModelFilter.php]: sure! I realised of that when you told me and it is already fixed

File: tests/Filter/NumberFilterTest.php [hunk 4]
```diff
@@ -54,50 +54,36 @@ public function testFilterInvalidOperator()
         $this->assertFalse($filter->isActive());
     }
 
-    public function testFilter()
+    /**
+     * @dataProvider getNumberExamples
+     */
+    public function testFilter(array $data, string $method)
```
Comment (phansys) [inline tests/Filter/NumberFilterTest.php]:
```suggestion
    public function testFilter(array $data, string $method): void
```

**Pull request merged**